ReleaseProcessAutomation.UnitTests/Steps/Releases/ReleaseRcStepTests.cs

```diff
@@ -104,8 +105,8 @@ public void Execute_callsNextVersionFromRelease_WithProperVersionCollection ()
     _gitClientStub.Setup(_ => _.IsOnBranch("release/")).Returns(true);
     _gitClientStub.Setup(_ => _.GetCurrentBranchName()).Returns("branchName");
     _gitClientStub.Setup(_ => _.DoesBranchExist("prerelease/v0.0.0")).Returns(false);
-    var nextPossibleVersions = nextVersion.GetNextPossibleVersionsDevelop();
-
+    
+    var nextPossibleVersions = nextVersion.GetNextPossibleVersionsReleaseBranchFromDevelop();
```

Comment:
Shouldn't the existing logic be tested in a separate unit test anyways? This should only check if it is using the correct extension method, if there is a bug with the extension method it should already be detected by the corresponding unit tests. 

Since we can't verify the correct call by using mocks, this is IMO the best option we have.

Reply:
yes and no (about "shouldn't it be tested anway?"). Any value that can be statically derived from the testsetup (and without the use of pen and paper and a one-time-pad) should still be hardcoded. Here, for instance, it seems from reading the test (not checked the prod-code) we're using the same method calls in both the test and production. This is typically a case of code duplication which could result in copy&paste errors (doing the wrong method call and copying it into the test as well) or other effects. It also hides what we actually try to do because you don't see the relevant information in plain string form but rather obfuscated behind API calls. 

For instance, if you do a call to string.Trim() or a call to string.Split(), the APIs you use are probably quite bugfree, but you can't be certain about the actual semantics. For instance, with Split(), you might pass a parameter that's incorrect and you don't notice because you have the same mistake in both places.

That said, I've looked at the test again and realize now, it's actually a list of versions you're getting, not just a single version and this result is used as a parameter in the setup-call. And the list might change in the future. Okay, yes, it's okay to do it like this here because inlining the result would pose a significant risk of future changes. Thanks for challangening :)

It also shows that the abstraction is possibly not perfect, a mockable service instead of an API on SemVersion might be the better choice. It's what you call "in hindsight" :D